lib/Raven/Serializer.php

```diff
@@ -67,17 +67,44 @@ public function __construct($mb_detect_order = null)
      * sanitization and encoding.
      *
      * @param mixed $value
+     * @param array $context
+     * @return string|bool|double|int|null|object|array
+     */
+    public function serialize($value, array $context = [])
+    {
+        $full_context = $this->get_full_context($context);
+        return $this->serialize_inner($value, $full_context['max_depth'], 0);
+    }
+
+    /**
+     * @param array $context
+     * @return array
+     */
+    protected function get_full_context(array $context)
+    {
+        if (!array_key_exists('max_depth', $context)) {
```

Comment:
>I want the situation where users can change serializers or can get them from Client instance

Users can change serializer through options, and this is not related in any way to getters. But the serializer is an internal thing used to convert values to something suitable to be sent to Sentry and there is no reason to expose it to the public. I don't want to be rude, but it seems to me you did not read anything of what I said in all my previously comments (this is the 3rd time I say this).